rocketpy/environment/tools.py

```diff
@@ -0,0 +1,684 @@
+    )
+
+
+def find_longitude_index(longitude, lon_list):
+    """Finds the index of the given longitude in a list of longitudes.
+
+    Parameters
+    ----------
+    longitude : float
+        The longitude to find in the list.
+    lon_list : list of float
+        The list of longitudes.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the adjusted longitude and its index in the list.
+
+    Raises
+    ------
+    ValueError
+        If the longitude is not within the range covered by the list.
+    """
+    # Determine if file uses -180 to 180 or 0 to 360
+    if lon_list[0] < 0 or lon_list[-1] < 0:
+        # Convert input to -180 - 180
+        lon = longitude if longitude < 180 else -180 + longitude % 180
+    else:
+        # Convert input to 0 - 360
+        lon = longitude % 360
+    # Check if reversed or sorted
+    if lon_list[0] < lon_list[-1]:
+        # Deal with sorted lon_list
+        lon_index = bisect.bisect(lon_list, lon)
+    else:
+        # Deal with reversed lon_list
+        lon_list.reverse()
+        lon_index = len(lon_list) - bisect.bisect_left(lon_list, lon)
+        lon_list.reverse()
+    # Take care of longitude value equal to maximum longitude in the grid
+    if lon_index == len(lon_list) and lon_list[lon_index - 1] == lon:
+        lon_index = lon_index - 1
+    # Check if longitude value is inside the grid
+    if lon_index == 0 or lon_index == len(lon_list):
+        raise ValueError(
+            f"Longitude {lon} not inside region covered by file, which is "
+            f"from {lon_list[0]} to {lon_list[-1]}."
+        )
+
+    return lon, lon_index
+
+
+def find_latitude_index(latitude, lat_list):
+    """Finds the index of the given latitude in a list of latitudes.
+
+    Parameters
+    ----------
+    latitude : float
+        The latitude to find in the list.
+    lat_list : list of float
+        The list of latitudes.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the latitude and its index in the list.
+
+    Raises
+    ------
+    ValueError
+        If the latitude is not within the range covered by the list.
+    """
+    # Check if reversed or sorted
+    if lat_list[0] < lat_list[-1]:
+        # Deal with sorted lat_list
+        lat_index = bisect.bisect(lat_list, latitude)
+    else:
+        # Deal with reversed lat_list
+        lat_list.reverse()
+        lat_index = len(lat_list) - bisect.bisect_left(lat_list, latitude)
+        lat_list.reverse()
+    # Take care of latitude value equal to maximum longitude in the grid
+    if lat_index == len(lat_list) and lat_list[lat_index - 1] == latitude:
+        lat_index = lat_index - 1
+    # Check if latitude value is inside the grid
+    if lat_index == 0 or lat_index == len(lat_list):
+        raise ValueError(
+            f"Latitude {latitude} not inside region covered by file, "
+            f"which is from {lat_list[0]} to {lat_list[-1]}."
+        )
+    return latitude, lat_index
```

Comment:
Once again. These lines are not new.
It would be nice to have different examples for a better coverage.
However, it is not a problem introduced by this PR